dlt/common/storages/fsspec_filesystem.py

```diff
@@ -185,7 +185,12 @@ def open(self, mode: str = "rb", **kwargs: Any) -> IO[Any]:  # noqa: A003
                 **text_kwargs,
             )
         else:
-            opened_file = self.fsspec.open(self["file_url"], mode=mode, **kwargs)
+            opened_file = self.fsspec.open(
+                self["file_url"],
+                mode=mode,
+                compression="gzip" if self["file_url"].endswith(".gz") else None,
```

Comment:
This line is the only actual change - set a compression flag for the `gz` files. Tried it locally with the csv_duckdb_reader - works fine

Reply:
ok this is good (for auto mode, but please use mime type (it is a part of FileDict) to detect compression type (does fsspec support anything else?)